### PR TITLE
[CELEBORN-275] WrappedCallback should only handle response from replica

### PR DESCRIPTION
**Pull request by boneanxs**
### What changes were proposed in this pull request?
1. Move `WrappedCallback` in the `doReplicate` block
2. Remove unnecessary check in the wrappedCallback
3. Currently slave push data timer metric is not well calculated, need to fix it

### Why are the changes needed?
Clean the codes and fix the bug of calculating slave push data timer

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?
MT. and existing UT.

### CODE REVIEW

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala [hunk 1]
```diff
@@ -385,11 +379,12 @@ class PushDataHandler extends BaseMessageHandler with Logging {
...
-    if (isMaster) {
-      workerSource.startTimer(WorkerSource.MasterPushDataTime, key)
-    } else {
-      workerSource.startTimer(WorkerSource.SlavePushDataTime, key)
-    }
+    val callbackWithTimer =
+      if (isMaster) {
+        new WrappedTimeMetric(workerSource, WorkerSource.MasterPushDataTime, key, callback)
+      } else {
+        new WrappedTimeMetric(workerSource, WorkerSource.MasterPushDataTime, key, callback)
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala:386]: SlavePushDataTime

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala [hunk 2]
```diff
@@ -645,6 +635,24 @@ class PushDataHandler extends BaseMessageHandler with Logging {
...
+    override def onSuccess(response: ByteBuffer): Unit = {
+      callback.onSuccess(response)
+      source.stopTimer(metricName, key)
+    }
+
+    override def onFailure(e: Throwable): Unit = {
+      callback.onFailure(e)
+      source.stopTimer(metricName, key)
+    }
+  }
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala:655]: please add a blank line after the class definition

**Review by waitinfuture — APPROVED** [review]
LGTM % some small comments, thanks!

**Review by AngersZhuuuu — APPROVED** [review]
LGTM except some minor comments

**Comment by AngersZhuuuu** [comment]
Pls run `./dev/reformat` to format the code

**Pull request merged**